### PR TITLE
Only change locality if borough is _not_ changed

### LINKED ISSUE #288: Enhance postal cities resilience against erroneous data
As it stands now, the postal cities dataset can cause records to have invalid admin hierarchy if even a _single_ record in OSM has an incorrect mapping from postal code to locality.

A good example of this is postal code `11215` in Brooklyn, NY, which currently shows up as part of [Geneseo, NY](https://spelunker.whosonfirst.org/id/85978297/), several hundred miles away.

[/v1/search?text=111+8th+Avenue%2C+Brooklyn%2C+Geneseo%2C+NY%2C+USA](https://pelias.github.io/compare/#/v1/search?text=111+8th+Avenue%2C+Brooklyn%2C+Geneseo%2C+NY%2C+USA&debug=0)
Screenshot_2020-03-06_13-42-18
image

It turns out there is a [single incorrect record in OSM](https://www.openstreetmap.org/way/375817791) with postal code `11215`:
Screenshot_2020-03-06_13-39-58

This is enough to introduce an incorrect mapping.

## Possible solutions

I'm sure there are many things we can do here, and we might end up including several:

- Require there to be more than one confirmation of any individual postal code <-> locality mapping, to guard against errors
- Improve our existing code that doesn't consider matches outside of a certain distance
- Add checks that can determine mapping outliers. In this case, for example, there are 130 OSM records that confirm the mapping from `11215` to `Brooklyn, NY`.

### PR DESCRIPTION
**Pull request by orangejulius**
Because we now manage postal cities replacements for both borough and locality values, we have to be a bit careful about when we replace each of them.

After investigating what looked like an error in our logic for determining the best postal cities match in
https://github.com/pelias/wof-admin-lookup/issues/288, it turns out a significant issue we were seeing in Brooklyn was actually a more subtle bug.

In short, it looks like if we replace the borough field with with a postal city value, we should _not_ replace the locality value.

Closes #288

### CODE REVIEW

**Comment by missinglink** [comment]
Hmm yeah it's odd that I went sofar as to add a code comment with this exact situation in the code but didn't consider this ;)

**Comment by orangejulius** [comment]
> Hmm yeah it's odd that I went sofar as to add a code comment with this exact situation in the code but didn't consider this ;)

Sometimes a problem guides you in a certain direction and there's usually a reason for that :)
